contracts/LinniaRecords.sol

```diff
@@ -55,87 +66,88 @@ contract LinniaRecords is Ownable {
     /* External functions */
 
     function addRecordByAdmin(
-        bytes32 fileHash, address fileOwner, address attestator,
-        string keywords, bytes32 ipfsHash)
+        bytes32 dataHash, address owner, address attestator,
+        string metadata, bytes32 dataUri)
         onlyOwner
         external
         returns (bool)
     {
-        require(
-            _addRecord(fileHash, fileOwner, keywords, ipfsHash)
-        );
+        bytes32 rootHash = _addRecord(dataHash, owner, metadata, dataUri);
+        require(rootHash != 0);
         if (attestator != 0) {
```

Comment:
Can you combine to make more readable.:
require(rootHash != 0 && attestator != 0)

Will this be less gas?

Reply:
Maybe this is also ok:

`require(rootHash != 0 && attestator != 0 && _addSig(rootHash, attestator))`